### ISSUE TITLE
Enumerated Types documentation

### ISSUE THREAD

**Issue by JohanMabille** [numpy/numpy#9974]
The current documentation of [enumerated types](https://docs.scipy.org/doc/numpy-dev/reference/c-api.dtype.html) makes me think there are 8 values possible for integers types ((signed, unsigned) * (8, 16, 32, 64) bits).

However, if I dump the values of the enumeration I have results that are not consistent with the documentation:

Windows x64 (with `sizeof(long) = 4`)
```bash
NPY_BYTE = 1
NPY_UBYTE = 2
NPY_INT8 = 1
NPY_UINT8 = 2
NPY_SHORT = 3
NPY_USHORT = 4
NPY_INT16 = 3
NPY_UINT16 = 4
NPY_INT = 5
NPY_UINT = 6
NPY_INT32 = 7
NPY_UINT32 = 8
NPY_LONG = 7
NPY_ULONG = 8
NPY_INT64 = 9
NPY_UINT64 = 10
NPY_LONGLONG = 9
NPY_ULONGLONG = 10
```
`NPY_INT` and `NPY_INT32` differ (the same for `NPY_UINT` and `NPY_UIN32`).

Linux x64 (with `sizeof(long) = 8`)
```bash
NPY_BYTE = 1
NPY_UBYTE = 2
NPY_INT8 = 1
NPY_UINT8 = 2
NPY_SHORT = 3
NPY_USHORT = 4
NPY_INT16 = 3
NPY_UINT16 = 4
NPY_INT = 5
NPY_UINT = 6
NPY_INT32 = 5
NPY_UINT32 = 6
NPY_LONG = 7
NPY_ULONG = 8
NPY_INT64 = 7
NPY_UINT64 = 8
NPY_LONGLONG = 9
NPY_ULONGLONG = 10
```
Here, `NPY_INT64` and `NPY_LONGLONG` differ (the same for `NPY_UIN64` and `NPY_ULONGLONG`)
In both cases (Windows and Linux) we have 10 values for integer instead of 8.

Is that expected ? I'm using numpy v1.13.3.

**Comment by eric-wieser** [comment]
Yep, this sucks, and we inherit it from C. There are 5 C-compiler-types of integer, each signed and unsigned:

* `np.byte`, `np.ubyte` (C `char`)
* `np.short`, `np.ushort` (C `short`)
* `np.intc`, `np.uintc` (C `int`)
* `np.int_`, `np.uint` (C `long`)
* `np.longlong`, `np.ulonglong` (C `long long`)

As you note though, typically these occupy only 4 different sizes, causing two of them to overlap.

**Comment by eric-wieser** [comment]
The docs are clear here:

> NPY_LONG
> > Equivalent to either NPY_INT or NPY_LONGLONG, depending on the platform.

The enumeration value may be different, but the behaviour is the same.
